File: CoreService/Models/Theme.cs
using System;
using System.Collections.Generic;

namespace CoreService.Models;

public partial class Theme
{
    public int Id { get; set; }

    public string Title { get; set; } = null!;

    public string Description { get; set; } = null!;

    public string? Tags { get; set; }

    public string? Department { get; set; }

    public bool Isarchived { get; set; }

    public string Suggestedby { get; set; } = null!;

    public string? Consultantname { get; set; }

    public string? Consultantcontact { get; set; }

    public int Supervisorid { get; set; }

    public DateTime Createddate { get; set; }

    public DateTime Updateddate { get; set; }

    public virtual ICollection<Practice> Practices { get; set; } = new List<Practice>();

    public virtual Lecturer Supervisor { get; set; } = null!;
}

File: frontend/src/entities/Theme.ts
import {Lecturer} from "./Lecturer.ts";
import {Consultant} from "./Consultant.ts";

export interface Theme {
    id: number;
    title: string;
    description: string;
    tags: string;
    level: string;
    department: string;
    isarchived: boolean;
    suggestedby: string;
    source: string;
    consultantid: number;
    supervisorid: number;
    createddate: Date;
    updateddate: Date;
    consultant: Consultant;
    supervisor: Lecturer;
}

export interface InputTheme {
    title: string;
    description: string;
    tags: string;
    level: string;
    department: string;
    isarchived: boolean;
    suggestedby: string;
    consultantid: number;
    supervisorid: number;
}